### DISCUSSION TITLE
HAOS - Nextcloud Office Setup (partially working atm)

### DISCUSSION THREAD

**Discussion by senna1992** [alexbelgium/hassio-addons#1502] [Q&A]
Hi folks,

i am trying to get my nextcloud instance enhanced by the use of its native office suite.
My Setup is based on the HA Yellow with an 8gb CM4 fitted.

Currently i managed to:
a) get the "collabora-code" server running inside a docker container,
b) have the server reachable for nextcloud instance (unstaple :( )

 
![fail](https://github.com/user-attachments/assets/4785abd8-5880-4bb3-ba33-abf789251c5b)
![success](https://github.com/user-attachments/assets/676f23d0-44f7-4ec1-8689-5d6b0eef8391)

I am currently using those addons / extensions to realize this project:

-> Portainer: setup new container with "collabora/code" image and have it configured as follows:
     IP: 172.30.33.10 ; Port Mapping: 9980 - 9980 (TCP); Network: hassio
-> Nextcloud App-Names: "Richdocuments" aka Nextcloud Office

-> Nginx Proxy manager: reverse proxy for mapping a dyndns domain, configured as follows:
     Mode: Http ; IP: 172.30.33.10 ; Port: 9980 ; Force SSL ; SSL Cert for the base domain selected

Till now, i got as far as to see the collabora's server version from within nextcloud's "admin" settings
This changes to "no connection" after a few seconds. 
Files dont open, as loading takes a few seconds and then crashes with error "Nextcloud office could not be loaded"

If someone knows, how to get this configured the right way - let me know and i will try =)

Hint: can't use the built in server, as the docker container of NC uses musl, which is not supported!

**Comment by alexbelgium** [comment]
Hi, alas I think it is by design, the linuxserver image on which the add-on is based auto removes it : https://github.com/linuxserver/docker-nextcloud?tab=readme-ov-file#collaborative-editing

**Reply by senna1992** [reply]
> Hi, alas I think it is by design, the linuxserver image on which the add-on is based auto removes it : https://github.com/linuxserver/docker-nextcloud?tab=readme-ov-file#collaborative-editing

Yea, this is for the built in server, which also is an option if you have non arm based server hardware.
My issue with "loosing" connection originates somewhere else, since i use another container with the appropriate arm64 image.

DIrectly installing the built in server as NC app - does work, but using wont. thats why i chose the more diffucult way



In other words, I just have the connector addon inside the nextcloud addon container.
The collabora/code container provides me with a joyful "unsupported software" greeting by HA, but for testing this is fine I guess  😁

**Reply by alexbelgium** [reply]
Sorry I read too fast :) I'm on holidays for the moment and will check in a couple days. Perhaps by looking in the add-on log, or running the occ command manually in the container wuthering portainer you would have more debug info (not guaranteed)

**Reply by senna1992** [reply]
It looks like some sort of ssl handshake issue, when accessing NC with enabled + connected collabora code container.
(makes no difference if i access NC using the domain or the IP)
![2024-08-03 06 53 04 192 168 0 42 53c9c95b5712](https://github.com/user-attachments/assets/ca82d6fc-982f-4912-90af-f6e083ec0ed8)

![2](https://github.com/user-attachments/assets/aa72d7cd-d76e-469e-b98c-faa2926bdd72)

**Comment by f4ntec** [comment]
Hi there,
I think we want to reach the same goal.
I'm using a Mini-PC (x86) with HAOS and Nextcloud installed as Add-On from this repository. NGINX is allready installed for external access including SSL. HA and NC are both available from External.

Now I wanted to use the built-in Office from Nextcloud. I'm not able to open Office-Files inside NC. Inside Administration of Nextcloud-Office-App, I receive the error 404
![Screenshot 2024-11-03 155603](https://github.com/user-attachments/assets/79d176c8-1149-4076-9a4b-b8d931b35e10)

@senna1992 you wrote that you fixed this error. Maybe you can tell me, what I need to do? Normally I use windows systems, so I only have minimal and rudimentary knowledges in linux. I've installed Studio Code Server on HA, I think this is helpful? 

NC is also telling me that I need to run some "occ" commands, but from Studio Code Server Terminal, these commands are unknown. Where I need to run them?

Thanks for your help!

**Reply by senna1992** [reply]
hi @f4ntec,

i did ditch this and went with onlyoffice and nextcloud running under "UmbrelOS" on my old thinkcentre m720t.
this way quite easy to do and also added another layer of possible services to host.
therefore i strongly suggest going some way like that over running this on a rasberry pi.

the fix you mentioned did not work reliablely across reboots. 
it was a dead end so i decided not to post any guides for it
